seckill-admin-service/src/main/java/io/servicecomb/poc/demo/seckill/web/SecKillAdminRestController.java

```diff
@@ -38,40 +42,79 @@
   private static final Logger logger = LoggerFactory.getLogger(SecKillAdminRestController.class);
 
   private final PromotionRepository promotionRepository;
+  private final SpringBasedPromotionEventRepository eventRepository;
 
   @Autowired
-  public SecKillAdminRestController(PromotionRepository promotionRepository) {
+  public SecKillAdminRestController(PromotionRepository promotionRepository,
+      SpringBasedPromotionEventRepository eventRepository) {
     this.promotionRepository = promotionRepository;
+    this.eventRepository = eventRepository;
   }
 
   @RequestMapping(method = RequestMethod.POST, value = "/")
   public ResponseEntity<String> create(@RequestBody PromotionDto promotionDto) {
     if (isValidPromotion(promotionDto)) {
-      Promotion promotion = new Promotion(promotionDto.getPublishTime(), promotionDto.getNumberOfCoupons(),
+      Promotion promotion = new Promotion(promotionDto.getPublishTime(), promotionDto.getFinishTime(),
+          promotionDto.getNumberOfCoupons(),
           promotionDto.getDiscount());
       promotionRepository.save(promotion);
       logger.info(
-          "Created a new promotion id = {}, number = {}, discount = {}, publishTime = {}",
+          "Created a new promotion id = {}, number = {}, discount = {}, publishTime = {}, finishTime = {}",
           promotion.getPromotionId(),
           promotion.getNumberOfCoupons(),
           promotion.getDiscount(),
-          promotion.getPublishTime());
+          promotion.getPublishTime(),
+          promotion.getFinishTime());
 
       return new ResponseEntity<>(promotion.getPromotionId(), OK);
     }
 
     return new ResponseEntity<>(String.format(
-        "Invalid promotion {coupons=%d, discount=%f, publishTime=%s}",
+        "Invalid promotion {numberOfCoupons=%d, discount=%f, publishTime=%s, finishTime=%s}",
         promotionDto.getNumberOfCoupons(),
         promotionDto.getDiscount(),
-        promotionDto.getPublishTime()),
+        promotionDto.getPublishTime(),
+        promotionDto.getFinishTime()),
+        BAD_REQUEST);
+  }
+
+  @RequestMapping(method = RequestMethod.PUT, value = "/")
+  public ResponseEntity<String> modify(@RequestBody PromotionDto promotionDto) {
```

Comment:
PUT needs pass resource id like `value="/{promotionId}"` and retrieve this id with `@PathVariable`